packages/ui/src/components/Schemas/TextSchema.tsx

```diff
@@ -27,14 +27,23 @@ const TextSchemaUI = (
 
   useEffect(() => {
     if (schema.dynamicFontSize && schema.data) {
-      calculateDynamicFontSize({ textSchema: schema, font, input: schema.data }).then(setDynamicFontSize);
+      calculateDynamicFontSize({
+        textSchema: schema,
+        font,
+        input: schema.data,
+        startingFontSize: dynamicFontSize,
```

Comment:
The idea of the previous change was initially to solve an issue of efficiency.

When calculating the dynamic font size it was based on an arbitrary value, which is the `schema.fontSize` before you check the dynamicFontSize checkbox.  This means that the font will be iteratively grown or shrunk from this arbitrary value on every change, which is very inefficient, especially as the strings get long.

My idea was to update the `schema.fontSize` with the dynamically calculated value so that the starting point should usually be very close the the new calculated value, which means less iterations in the calculation code.

This alternative passes in a `startingFontSize` which is what we are currently using as the dynamic font size - so the end result for the UI should be the same - better performance.